apps/rush/UPGRADING.md

```diff
@@ -0,0 +1,51 @@
+# Upgrade notes for @microsoft/rush
+
+### Rush 5.60.0
+
+This release of Rush changed the schema of the `<project-root>/config/rush-project.json` file.
+The new schema introduces a concept of "operations" and an `operationSettings` property. An operation
+is a command or phase that is invoked in a project. The top-level `projectOutputFolderNames` property
+has been removed in favor of a per-operation `outputFolderNames` property. The `phaseOptions` and
+`buildCacheOptions` properties have also been removed in favor of a per-operation properties.
+Below is an example diff between an old and new `rush-project.json` file:
+
+```diff JSON
+{
+  "incrementalBuildIgnoredGlobs": ["temp/**"],
+-  "projectOutputFolderNames": ["output-folder-1", "output-folder-2"],
+-  "phaseOptions" [
+-    {
+-      "phaseName": "_phase:build",
+-      "projectOutputFolderNames": ["output-folder-a", "output-folder-b"]
+-    }
+-  ]
+-  "buildCacheOptions": {
+-    "disableBuildCache": false,
+-    "optionsForCommands" [
+-      {
+-        "commandName": "test",
+-        "disableBuildCache": true
+-      }
+-    ]
+-  }
++  "disableBuildCache": false,
++  "operationSettings": [
++    {
++      "operationName": "build",
++      // The "build" operation's output folder names were previously defined
++      // in the top-level `projectOutputFolderNames` property.
++      "outputFolderNames": ["output-folder-1", "output-folder-2"]
++    },
++    {
++      "operationName": "_phase:build",
++      "outputFolderNames": ["output-folder-a", "output-folder-b"]
++    },
++    {
++      "operationName": "test",
++      "disableBuildCacheForOperation": true
++    }
++  ]
+}
+```
```

Comment:
🤔 I think this might be more friendly as two complete files like we did in [heft/UPGRADING.md](https://github.com/microsoft/rushstack/blob/7ffc0ee518535348fd1804100af6ebef048e72b4/apps/heft/UPGRADING.md):


``````markdown
**rush-project.json** (OLD)
```js
{
  "incrementalBuildIgnoredGlobs": ["temp/**"],
  "projectOutputFolderNames": ["output-folder-1", "output-folder-2"],
  "phaseOptions": [
    {
      "phaseName": "_phase:build",
      "projectOutputFolderNames": ["output-folder-a", "output-folder-b"]
    }
  ],
  "buildCacheOptions": {
    "disableBuildCache": false,
    "optionsForCommands": [
      {
        "commandName": "test",
        "disableBuildCache": true
      }
    ]
  }
}
```

**rush-project.json** (NEW)
```js
{
  "incrementalBuildIgnoredGlobs": ["temp/**"],

  "disableBuildCache": false,  // formerly buildCacheOptions.disableBuildCache

  "operationSettings": [  // formerly phaseOptions
    {
      "operationName": "build",

      // The "build" operation's output folder names were previously defined
      // in the top-level `projectOutputFolderNames` property.
      "outputFolderNames": ["output-folder-1", "output-folder-2"]
    },
    {
      "operationName": "_phase:build",  // formerly phaseName
      "outputFolderNames": ["output-folder-a", "output-folder-b"]
    },
    {
      "operationName": "test",
      "disableBuildCacheForOperation": true
    }
  ]
}
```
``````